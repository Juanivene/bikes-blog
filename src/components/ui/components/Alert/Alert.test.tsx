import { fireEvent, render, screen } from '@testing-library/react';

import Alert from './Alert';

describe('Alert Component', () => {
  let renderComponent: HTMLElement;

  beforeEach(() => {
    // Renderizamos el componente una sola vez antes de cada prueba
    const renderResult = render(<Alert>Default Alert</Alert>);
    renderComponent = renderResult.container;
  });

  // Prueba de snapshot
  it('snapshot', () => {
    expect(renderComponent).toMatchSnapshot();
  });

  // Verificar renderizado con props por defecto
  it('renders the alert with default props', () => {
    const alert = screen.getByText('Default Alert');
    expect(alert).toBeInTheDocument();
  });

  // Verificar el cierre del componente
  it('closes the alert when the close button is clicked', () => {
    const { container } = render(<Alert closable>Closable Alert</Alert>);
    const closeButton = screen.getByRole('button', { name: /cerrar/i });
    fireEvent.click(closeButton);
    expect(container.firstChild).toHaveClass('hidden');
  });

  // Verificar los diferentes tipos de alerta
  // it('renders the correct icon for each alert type', () => {
  //   const types = ['error', 'info', 'success', 'warning', 'loading'] as const;

  //   types.forEach((type) => {
  //     render(<Alert type={type}>Alert with {type} type</Alert>);
  //     const alert = screen.getByText(`Alert with ${type} type`);
  //     expect(alert).toBeInTheDocument();

  //     // Check the correct background class for each type
  //     expect(renderComponent.firstChild).toHaveClass(
  //       type === 'loading'
  //         ? 'bg-gray-100 dark:bg-gray-800'
  //         : `bg-${type}-200 dark:bg-${type}-800`
  //     );

  //     // Check if the correct icon is rendered
  //     if (type === 'loading') {
  //       expect(screen.getByRole('status')).toBeInTheDocument();
  //     } else {
  //       const icon = screen.getByTestId(`alert-icon-${type}`);
  //       expect(icon).toBeInTheDocument();
  //     }
  //   });
  // });

  // Verificar que no se renderiza el Ã­cono si hideIcon es true
  // it('does not render an icon when hideIcon is true', () => {
  //   render(<Alert hideIcon>No Icon Alert</Alert>);
  //   const icon = screen.queryByRole('img');
  //   expect(icon).not.toBeInTheDocument();
  // });

  // Verificar que se renderiza con la clase adecuada
  it('applies the correct classes based on the type', () => {
    const { container } = render(<Alert type="error">Error Alert</Alert>);
    expect(container.firstChild).toHaveClass('bg-red-200', 'dark:bg-red-800');
  });

  // Verificar el ancho del texto dependiendo de las props
  // it('calculates the correct text width based on props', () => {
  //   const { container: defaultContainer } = render(<Alert>Default Alert</Alert>);
  //   expect(defaultContainer.querySelector('[data-testid="alert"]')).toHaveClass('xs:12');

  //   const { container: closableContainer } = render(<Alert closable>Closable Alert</Alert>);
  //   expect(closableContainer.querySelector('[data-testid="alert"]')).toHaveClass('xs:11');

  //   const { container: hideIconContainer } = render(<Alert hideIcon>Hide Icon Alert</Alert>);
  //   expect(hideIconContainer.querySelector('[data-testid="alert"]')).toHaveClass('xs:11');
  // });

  // Verificar accesibilidad general
  // it('is accessible', () => {
  //   render(<Alert aria-label="Accessible Alert">Accessible Alert</Alert>);
  //   const alert = screen.getByLabelText('Accessible Alert');
  //   expect(alert).toBeInTheDocument();
  // });

  // Verificar que se oculta si open es false
  // it('is hidden when open is false', () => {
  //   const { container } = render(<Alert>Hidden Alert</Alert>);
  //   fireEvent.click(screen.getByRole('button', { name: /cerrar/i }));
  //   expect(container.firstChild).toHaveClass('hidden');
  // });
});
