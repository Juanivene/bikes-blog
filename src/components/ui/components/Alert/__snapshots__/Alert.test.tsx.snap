// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert Component > snapshot 1`] = `
<div>
  <div
    class="grid grid-cols-12 row-span-1 bg-sky-200 dark:bg-sky-800 rounded-xl px-2 py-3 dark:text-white md:pr-3"
  >
    <div
      class="col-span-2 sm:col-span-1 justify-center flex md:mr-1"
    >
      <div>
        <svg
          fill="currentColor"
          height="1.5em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1.5em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M11 7h2v2h-2V7zm0 4h2v6h-2v-6zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
          />
        </svg>
      </div>
    </div>
    <div
      class="col-span-11"
    >
      Default Alert
    </div>
  </div>
</div>
`;
